examples/example4-debuggingtools.py

```diff
@@ -75,15 +76,57 @@
 
 # --- Probe ----
 
-# now that we have built some stuff, let's clear it so we can try again in a
+# Now that we have built some stuff, let's clear it so we can try again in a
 # different way.  We can start by clearing all of the hardware from the current working
 # block.  The working block is a global structure that keeps track of all the
 # hardware you have built thus far.  A "reset" will clear it so we can start fresh.
-# pyrtl.reset_working_block()
+pyrtl.reset_working_block()
+print("---- Using Probes ----")
+
+# In this example, we will be multiplying two numbers using tree_multiplier()
+# Again, create the two inputs and an output
+in1, in2 = (pyrtl.Input(8, "in" + str(x)) for x in range(1, 3))
+
+multout = multipliers.tree_multiplier(in1, in2)
+
+# The following line will create a probe named 'stdout_probe" for later use, like an output.
+# We could also use it during assignment like "out <<= pyrtl.probe(multout, 'stdout_probe')"
+# if 'out' were an Output.
+pyrtl.probe(multout, 'stdout_probe')
+# probe can also be used with other operations like this:
+pyrtl.probe(multout + 32, 'adderout_probe')
+# or this:
+pyrtl.probe(multout[2:7], 'selectout_probe')
+# or, similarly:
+pyrtl.probe(multout)[2:16]  # notice probe names are not absolutely necessary
```

Comment:
Pretty sure that's all of them, it's just that they aren't used mid-assignment because I thought it would be important to show that Outputs can be largely avoided by using probes, when simulating simple things like an adder.